.pre-commit-config.yaml

```diff
@@ -0,0 +1,49 @@
+ci:
+  autofix_prs: true
+  autoupdate_schedule: weekly
+  autoupdate_commit_msg: 'chore: pre-commit autoupdate'
+
+repos:
+- repo: https://github.com/pre-commit/pre-commit-hooks
+  rev: v4.4.0
+  hooks:
+  - id: check-json
+  - id: check-toml
+  - id: check-yaml
+  - id: end-of-file-fixer
+    exclude: (docs/.*|samples/.*\.json)
+  - id: trailing-whitespace
+    exclude: |
+        (?x)^(
+            .bumpversion.cfg
+        )$
+
+- repo: https://github.com/psf/black
+  rev: 23.1.0
+  hooks:
+  - id: black
+
+- repo: https://github.com/pycqa/isort
+  rev: 5.12.0
+  hooks:
+  - id: isort
+
+- repo: https://github.com/asottile/pyupgrade
+  rev: v3.3.1
+  hooks:
+  - id: pyupgrade
+    args: ['--py37-plus']
+
+- repo: https://github.com/charliermarsh/ruff-pre-commit
+  rev: v0.0.255
+  hooks:
+  - id: ruff
+    args: [--fix, --exit-non-zero-on-fix]
```

Comment:
The Ruff docs recommend putting this before black and isort: https://beta.ruff.rs/docs/usage/#using-ruff